clients/src/test/java/org/apache/kafka/clients/NetworkClientTest.java

```diff
@@ -126,18 +158,44 @@ private void checkSimpleRequestResponse(NetworkClient networkClient) {
         assertEquals("Should be correlated to the original request", request.header(), handler.response.requestHeader());
     }
 
-    private void awaitReady(NetworkClient client, Node node) {
+    private void maybeSetExpectedApiVersionsResponse() {
+        final Collection<ApiVersionsResponse.ApiVersion> expectedApiVersions = getExpectedApiVersions();
+        if (expectedApiVersions == null)
+            return;
+        ResponseHeader respHeader = new ResponseHeader(0);
+        Struct resp = new Struct(ProtoUtils.currentResponseSchema(ApiKeys.API_VERSIONS.id));
+        resp.set(ERROR_CODE_KEY_NAME, (short) 0);
+        List<Struct> apiVersionList = new ArrayList<>();
+        for (ApiVersionsResponse.ApiVersion apiVersion : expectedApiVersions) {
+            Struct apiVersionStruct = resp.instance(API_VERSIONS_KEY_NAME);
+            apiVersionStruct.set(API_KEY_NAME, apiVersion.apiKey);
+            apiVersionStruct.set(MIN_VERSION_KEY_NAME, apiVersion.minVersion);
+            apiVersionStruct.set(MAX_VERSION_KEY_NAME, apiVersion.maxVersion);
+            apiVersionList.add(apiVersionStruct);
+        }
+        resp.set(API_VERSIONS_KEY_NAME, apiVersionList.toArray());
+        int size = respHeader.sizeOf() + resp.sizeOf();
+        ByteBuffer buffer = ByteBuffer.allocate(size);
+        respHeader.writeTo(buffer);
+        resp.writeTo(buffer);
+        buffer.flip();
```

Comment:
We should have some shared code for serialising the response and response header instead of duplicating it in multiple places.